tests/test_echo.py

```diff
@@ -2,11 +2,142 @@
+        args_upper = process.args
+        stdout, _ = process.communicate()
+        stdout_upper = stdout.decode("utf-8")
+
+        usage = open("./USAGEUPPER", "r").read()
+        self.assertEquals(stdout_u, usage)
+        self.assertEqual("-u" in args_u, True)
+        self.assertEquals(stdout_upper, usage)
+        self.assertEqual("--upper" in args_upper, True)
+
+    def test_lower(self):
+        """Running the program with -l or --lower as arguments should result with
+        "lower" stored in namespace returned from parser.parse_args and that
+        "Hello" will return to "hello" when the program is run"""
+
+        process = subprocess.Popen(
+            ["python", "./echo.py", "-l", "Hello"],
+            stdout=subprocess.PIPE)
+        stdout, _ = process.communicate()
+        args_l = process.args
+        stdout_l = stdout.decode("utf-8")
+
+        process = subprocess.Popen(
+            ["python", "./echo.py", "--lower", "Hello"],
+            stdout=subprocess.PIPE)
+        args_lower = process.args
+        stdout, _ = process.communicate()
+        stdout_lower = stdout.decode("utf-8")
+
+        usage = open("./USAGELOWER", "r").read()
+        self.assertEquals(stdout_l, usage)
+        self.assertEqual("-l" in args_l, True)
+        self.assertEquals(stdout_lower, usage)
+        self.assertEqual("--lower" in args_lower, True)
+
+    def test_title(self):
+        """Running the program with -t or --title as arguments should result with
+        "title" stored in namespace returned from parser.parse_args and that
+        "hello" will return to "Hello" when the program is run"""
+
+        process = subprocess.Popen(
+            ["python", "./echo.py", "-t", "hello"],
+            stdout=subprocess.PIPE)
+        args_t = process.args
+        stdout, _ = process.communicate()
+        stdout_t = stdout.decode("utf-8")
+
+        process = subprocess.Popen(
+            ["python", "./echo.py", "--title", "hello"],
+            stdout=subprocess.PIPE)
+        args_title = process.args
+        stdout, _ = process.communicate()
+        stdout_title = stdout.decode("utf-8")
+
+        usage = open("./USAGETITLE", "r").read()
+        self.assertEquals(stdout_t, usage)
+        self.assertEqual("-t" in args_t, True)
+        self.assertEquals(stdout_title, usage)
+        self.assertEquals("--title" in args_title, True)
+
+    def test_all(self):
+        """Running the program with all three arguments should result with
+        the first argument stored in namespace returned from
+        parser.parse_args and that the first argument's case will be applied
+        to the returned string"""
+
+        process = subprocess.Popen(
+            ["python", "./echo.py", "-t", "-u", "-l", "hello"],
+            stdout=subprocess.PIPE)
+        stdout, _ = process.communicate()
+        stdout_t = stdout.decode("utf-8")
+        usage_t = open("./USAGETITLE", "r").read()
+        self.assertEquals(stdout_t, usage_t)
```

Comment:
as i understand the intent of the program,` $python echo.py "hello" -ult` should return the string:
```
HELLO
hello
Hello
```

The program as you wrote it doesn't look like it'd do that, and these tests reflect the program as you wrote it.  You may want to take another look at this.

Reply:
Step 6: Test for when all options are provided
When a user provides all three options, they should be applied in the order listed in the helpful usage message that Argparse constructs from the argument definitions. Here are a few examples:

$ python echo.py -tul "heLLo!"
Hello!
$ python echo.py -ul "heLLo!"
hello!
Note that the order that the options are provided doesn't matter, e.g. '-tul' and '-utl' and '-lut' are all equivalent inputs to Argparse. Only the final text transform result should be printed.